examples/sampling-integrated-gaussian-log-likelihood.ipynb

```diff
@@ -0,0 +1,542 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Integrating out the noise in Gaussian likelihood problems\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "A property of Gaussian likelihoods and certain is that the joint posterior distribution $p(\\theta,\\sigma)$ can be analytically integrated with respect to $\\sigma$ to yield the marginal distribution $p(\\theta)$. For example, for Gaussian log-likelihood models where we assume a prior, $\\sigma\\sim U(a, b),$ then the marginal posterior is given by,\n",
+    "\n",
+    "$p(\\theta) \\propto \\frac{\\pi ^{-n/2} \\text{sse}(\\theta)^{\\frac{1}{2}-\\frac{n}{2}} \\left[\\Gamma\n",
+    "   \\left(\\frac{n-1}{2},\\frac{\\text{sse}(\\theta)}{2 b^2}\\right)-\\Gamma\n",
+    "   \\left(\\frac{n-1}{2},\\frac{\\text{sse}(\\theta)}{2 a^2}\\right)\\right]}{2 \\sqrt{2} (b - a)}$,\n",
+    "   \n",
+    "where $\\text{sse}(\\theta) = \\sum_{i=1}^n(f_i(\\theta) - y_i)^2,$ is the sum of square errors and $\\Gamma(u,v)$ is the upper incomplete gamma function.\n",
+    "\n",
+    "In this notebook, we illustrate how using the posterior geometry where the noise parameter $\\sigma$ is integrated out can lead to faster convergence of the posterior distribution than the circumstance where $\\sigma$ is estimated.\n",
+    "\n",
+    "An issue with using the integrated log-likelihood is that the estimated model is no longer generative and so posterior predictive checking is harder. Whether this is more than compensated for by the speed up of sampling from a posterior of lower dimensionality is problem specific although unlikely to considerably speed up sampling."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Single output problem - logistic model\n",
+    "Here we illustrate how a posterior distribution for the logistic model growth rate and carrying capacity parameters can be estimated using both full posterior and the marginal posterior geometries."
```

Comment:
The final two chains Fitzhugh-Nagumo with and without this method don't look that different to me. Both seem to start mixing around 1000 iterations and have an R-hat of approx 1.00xyz ?

Reply:
It varies from run to run but they tend to be slightly faster converging in the latter case...have added a note in the example notebook.

Reply:
@MichaelClerx I've also added a section to the notebook where we compare effective sample size -- we get much better ess from the marginal sampler (that is, the Gaussian integrated one)!